lib/Diagnostics/DiagnosticsAlertsAnalyzer.js

```diff
@@ -179,11 +174,14 @@ class EntryDiagnostics {
                     }
                 }
             });
+
+            if (this.alertsToSend.length) {
+                this.emit('reportBeacon', this.alertsToSend);
```

Comment:
maybe you should also have this.alertsToRemove..., so this will solve the other problem...